ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -18,108 +30,227 @@
+    VungleInitializer.getInstance().initialize(config.getAppId(), context.getApplicationContext(),
+        new VungleInitializer.VungleInitializationListener() {
+          @Override
+          public void onInitializeSuccess() {
+            loadBanner();
+          }
+
+          @Override
+          public void onInitializeError(String errorMessage) {
+            Log.d(TAG, "SDK init failed: " + VungleBannerAdapter.this);
+            VungleManager.getInstance().removeActiveBannerAd(mPlacementId);
+            VungleListener listener = mVungleListener.get();
+            if (mPendingRequestBanner && listener != null) {
+              listener.onAdFailedToLoad(AdRequest.ERROR_CODE_INTERNAL_ERROR);
+            }
+          }
+        });
   }
 
-  // Use weak references to allow VungleInterstitialAdapter to be garbage collected.
-  // Banner view need to be added/removed on adLayout's onAttachedToWindow/onDetachedFromWindow
-  // to break view's parent-child references chain to the leaked VungleBannerAdapter in
-  // VungleManager.
-  void setAdLayout(@NonNull RelativeLayout adLayout) {
-    this.mAdLayout = new WeakReference<>(adLayout);
+  View getBannerView() {
+    Log.d(TAG, "getBannerView # instance: " + hashCode());
+    return adLayout;
   }
 
-  void setVungleListener(@Nullable VungleListener vungleListener) {
-    this.mVungleListener = new WeakReference<>(vungleListener);
+  public void onDestroy() {
+    Log.d(TAG, "onDestroy: " + hashCode());
+    destroy(adLayout);
+    adLayout = null;
   }
 
-  @Nullable
-  private VungleListener getVungleListener() {
-    return mVungleListener.get();
+  public void onPause() {
+    Log.d(TAG, "onPause");
+    updateVisibility(false);
   }
 
-  boolean isActive() {
-    return mAdLayout.get() != null;
+  public void onResume() {
+    Log.d(TAG, "onResume");
+    updateVisibility(true);
   }
 
-  void requestBannerAd(@NonNull Context context, @NonNull String appId) {
-    Log.d(TAG, "requestBannerAd: " + this);
-    mPendingRequestBanner = true;
-    VungleInitializer.getInstance()
-        .initialize(
-            appId,
-            context.getApplicationContext(),
-            new VungleInitializer.VungleInitializationListener() {
-              @Override
-              public void onInitializeSuccess() {
-                loadBanner();
-              }
-
-              @Override
-              public void onInitializeError(String errorMessage) {
-                Log.d(TAG, "SDK init failed: " + VungleBannerAdapter.this);
-                VungleListener listener = getVungleListener();
-                mVungleManager.removeActiveBannerAd(mPlacementId);
-                if (mPendingRequestBanner && listener != null) {
-                  listener.onAdFailedToLoad(AdRequest.ERROR_CODE_INTERNAL_ERROR);
-                }
-              }
-            });
+  @Nullable
+  String getUniquePubRequestId() {
+    return mUniquePubRequestId;
+  }
+
+  boolean isActive() {
+    return adLayout != null;
```

Comment:
adLayout = null is called only in onDestroy(), which is not guaranteed. Banner will newer became "unActive"

It breaks AND-2722, the main idea of which was to hold a weak reference in the Banner adapter to catch the moment when InterestialAdapter is garbage collected.
We need to move adLayout back to Interestial adapter or pass some another reference from Interestial adapter.